spec/01-unit/26-tracing/02-propagation_spec.lua

```diff
@@ -799,6 +899,15 @@ describe("propagation.set", function()
       )
     }
 
+    local gcp_headers 
+    if tonumber(span_id, 10) then
+      -- strip leading 0s in expected output 
+      gcp_headers = {["x-cloud-trace-context"] = gcp_trace_id .. "/" .. tonumber(span_id, 10) .. ";o=1"}
+    else
+      -- hex values are not valid span id inputs, translate to decimal
+      gcp_headers = {["x-cloud-trace-context"] = gcp_trace_id .. "/" .. openssl_bignumber.from_hex(span_id):to_dec() .. ";o=1"}
+    end
```

Comment:
I needed to detect whether the span id was a decimal or hex value because the span id part of a gcp trace header is written as a decimal value.  using `tonumber(span_id, 16)` on the hex value was resulting in a span_id number like `1.844674407371e+19` instead of `18446744073709551603`, which is why I ended up using the openssl library to do the conversion.

I _think_ this is an ok way to handle this and all tests were passing locally 🤷 .  As I mentioned though I'm pretty new to Lua, please let me know if there's a better way to handle this!

Reply:
that's **great**, good job!
You can save a couple of lines though and consider all span IDs to be hex, since span_id == "0000000000000003" == 0x3 the result will be the same.

```suggestion
    -- hex values are not valid span id inputs, translate to decimal
    local gcp_headers = {["x-cloud-trace-context"] = gcp_trace_id .. "/" .. openssl_bignumber.from_hex(span_id):to_dec() .. ";o=1"}
```

Reply:
oh 🤦 yeah good catch.  I've included that updated in the latest push